source/convection_diffusion/assemble_rhs.cc

```diff
@@ -263,51 +390,55 @@ void HeatEquation<dim>::assemble_local_rhs
     for (unsigned int i = 0; i < scratch.dofs_per_cell; ++i)
     {
       // Local right hand side (Domain integrals)
-      data.local_rhs(i) -=
-          (scratch.grad_phi[i] *
-           diffusion_term[q]
-           -
-           scratch.phi[i] *
-           (heat_source_term[q]
-            -
-            explicit_temperature_term[q]
-            -
-            advection_term[q])) *
-          scratch.temperature_fe_values.JxW(q);
+      data.local_rhs(i) +=
+          (-diffusion_term[q] * scratch.grad_phi[i] +
+           (source_term[q] - explicit_temperature_term[q] - advection_term[q]
+           ) * scratch.phi[i]
+          ) * scratch.temperature_fe_values.JxW(q);
```

Comment:
Somewhat of a convention that I have used throughout the assembly is to always chose the test function as the first term of the scalar contraction. Following the good practice of always multiplying the PDE with the test function from the left side.